File: ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here

  string = str.split('')
  index = string.length
  result = []

  

  

index.times do
    result << string[index - 1]
    index -= 1
  end

  return result.join('')

  
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

# initialize a string variable, storing the split of the str
# initialize a index variable, storing the string length
# initialize an empty array
# a times for loop that will run value store in the index variable, 
# and will run the line where the string[index - 1] will be appended into the array
# then return the result array with a join('')

File: javascript/reverse_string.js
function reverseString(str) {
  // type your code here

  let result = [];

  for (let i = str.length - 1; i > -1; i--) {
    result.push(str[i]);

  }

  return result.join("");


}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));

  console.log("Expecting: 'auhsoJ'");
  console.log("=>", reverseString("Joshua"));

  console.log("");

  console.log("Expecting: 'Buggie'");
  console.log("=>", reverseString("eigguB"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

// Initialize an empty array called result
// iterate through each letter in str
// push each letter of the string into the result array
// return result.join("")
